python/lsst/meas/algorithms/reinterpolate_pixels.py

```diff
@@ -0,0 +1,140 @@
+# This file is part of meas_algorithms.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("ReinterpolatePixelsConfig", "ReinterpolatePixelsTask")
+
+
+import math
+
+import lsst.afw.detection as afwDetection
+import lsst.afw.math as afwMath
+import lsst.meas.algorithms as measAlg
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.afw.image import MaskedImageF
+
+
+class ReinterpolatePixelsConfig(pexConfig.Config):
+    """Config for ReinterpolatePixelsTask"""
+
+    kernelFwhm = pexConfig.Field[float](
+        doc="FWHM of double Gaussian smoothing kernel.",
+        default=1.0,
+    )
+
+    growFootprints = pexConfig.Field[int](
+        doc="Number of pixels to grow footprints for pixels.",
+        default=1,
+    )
+
+    maskNameList = pexConfig.ListField[str](
+        doc="Mask plane name.",
+        default=["SAT"],
+    )
+
+    fallbackValue = pexConfig.Field[float](
+        doc="Value of last resort for interpolation.",
+        default=None,
+        optional=True,
+    )
+
+
+class ReinterpolatePixelsTask(pipeBase.Task):
+    """Task to reinterpolate pixels"""
+
+    ConfigClass = ReinterpolatePixelsConfig
+    _DefaultName = "reinterpolatePixels"
+
+    def run(self, exposure):
+        """Reinterpolate pixels over ``exposure``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to interpolate over.
+
+        Notes
+        -----
+        ``exposure`` is modified in place and will become the reinterpolated exposure.
+        """
+        mask = exposure.mask
+
+        if self.config.growFootprints > 0 and "SAT" in self.config.maskNameList:
+            # If we are interpolating over an area larger than the original
+            # masked region, we need to expand the original mask bit to the
+            # full area to explain why we interpolated there.
+            from lsst.ip.isr import growMasks
+
+            growMasks(mask, radius=self.config.growFootprints, maskNameList=["SAT"], maskValue="SAT")
```

Comment:
I think this is going to leave those mask planes grown as a side-effect, and I don't think we want that.

If this growing-as-a-side-effect behavior is also what happens in ISR, that might mean we don't need to have the capability to grow at all here, and could remove that config option and the logic that goes with it.

I haven't spotted that as being the case, but I didn't look very carefully, and I'm curious what the motivation for this grow logic was.  If we can get away without the grow here that'd be cleaner, but if there's a reason we expect to need it, we should:
- reverse it after interpolating;
- make the mask planes grown configurable.

Reply:
Regarding the motivation for the grow logic, I'm not entirely certain. I followed the logic in `interpolateFromMask()` from `ip_isr` [here](https://github.com/lsst/ip_isr/blob/c3c1122503caa670758387223c53d247d18482f3/python/lsst/ip/isr/isrFunctions.py#L184), where it appears that the grow logic is implemented to ensure that the interpolation covers an expanded region, including the original masked area, when dealing with *saturated pixels*. However, the ticket description suggests that this task needs to be made more reusable without relying on ISR-specific logic.

Reply:
In `interpolateFromMask()`, it appears that the mask planes remain grown and are not reversed back to their original state after the interpolation process.

Reply:
Growing is done for two reasons.   In the column direction you need to grow by one pixel as charge can spill from the last pixel over saturation into its neighbour away from the bright star.  In the row direction, we saw problems with CTE in the serial and "fixed" that by growing in the column direction.   The latter shouldn't be necessary for LSSTCam

Reply:
Looks like we can assume the SAT mask has already been grown, so there is no need to do it again. However, @natelust suggested keeping the `growFootprints` for SAT mask as a config option (will rename it to `growSaturatedFootprints` for specificity), with a default value of 0. This way, if someone wants to expand the mask even further for some reason, they can. But in that case, I will not be leaving the SAT mask grown after the reinterplation finishes - I'll simply revert this extra mask expansion at the end of the `run()` method.

Reply:
Also, I chatted with @czwa, and he pretty much agrees with Robert's response: "We intentionally grow saturated pixels because they tend to spill over to their neighboring pixels."